### PR TITLE
SEOImages: User Story 1463880, Part 31

### PR DESCRIPTION
**Pull request by damabe**
## Summary

Updating image file names and references to improve SEO.

Fixes #1463880

Contributes to #10839

### CODE REVIEW

**Comment by damabe** [comment]
@mairaw This PR is ready for review. CATS report: https://contentqacats.azurewebsites.net/CheckResults/2019032000095/Report/

There are 9 image names identified by CATS as being too short. They should be handled at a later time as they are not part of the SEO image without hyphen work.

File: docs/csharp/programming-guide/concepts/linq/filtering-data.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ Filtering refers to the operation of restricting the result set to contain only
   
  The following illustration shows the results of filtering a sequence of characters. The predicate for the filtering operation specifies that the character must be 'A'.  
   
- ![LINQ Filtering Operation](../../../../csharp/programming-guide/concepts/linq/media/linq_filter.png "LINQ_Filter")  
+ ![Diagram that shows a LINQ Filtering Operation](./media/filtering-data/linq-filter-operation.png)
```
Comment (mairaw) [inline docs/csharp/programming-guide/concepts/linq/filtering-data.md]:
```suggestion
 ![Diagram that shows a LINQ filtering operation](./media/filtering-data/linq-filter-operation.png)  
```

File: docs/framework/data/adonet/entity-data-model-inheritance.md [hunk 2]
```diff
@@ -13,7 +13,7 @@ The Entity Data Model (EDM) supports inheritance for [entity types](../../../../
 ## Example  
  The diagram below shows a conceptual model with four entity types: `Book`, `FictionBook`, `Publisher`, and `Author`. The `FictionBook` entity type is a derived type, inheriting from the `Book` entity type. The `FictionBook` type inherits the `ISBN (Key)`, `Title`, and `Revision` properties, and defines an additional property called `Genre`.  
   
- ![Inheritance](../../../../docs/framework/data/adonet/media/inheritanceexample.gif "InheritanceExample")  
+ ![Diagram that shows a model with four entity types.](./media/entity-data-model-inheritance/entity-type-inheritance.gif)
```
Comment (mairaw) [inline docs/framework/data/adonet/entity-data-model-inheritance.md]:
```suggestion
 ![Diagram that shows a conceptual model with four entity types.](./media/entity-data-model-inheritance/entity-type-inheritance.gif)  
```

File: docs/framework/data/adonet/entity-data-model-inheritance.md [hunk 3]
```diff
@@ -13,7 +13,7 @@ The Entity Data Model (EDM) supports inheritance for [entity types](../../../../
 ## Example  
  The diagram below shows a conceptual model with four entity types: `Book`, `FictionBook`, `Publisher`, and `Author`. The `FictionBook` entity type is a derived type, inheriting from the `Book` entity type. The `FictionBook` type inherits the `ISBN (Key)`, `Title`, and `Revision` properties, and defines an additional property called `Genre`.
```
Comment (mairaw) [inline docs/framework/data/adonet/entity-data-model-inheritance.md]:
```suggestion

The following diagram shows a conceptual model with four entity types: `Book`, `FictionBook`, `Publisher`, and `Author`. The `FictionBook` entity type is a derived type, inheriting from the `Book` entity type. The `FictionBook` type inherits the `ISBN (Key)`, `Title`, and `Revision` properties, and defines an additional property called `Genre`.  
```

File: docs/framework/wcf/diagnostics/tracing/emitting-user-code-traces.md [hunk 4]
```diff
@@ -122,13 +122,11 @@ List of activities by creation time (left panel) and their nested activities (up
   
  If the service code throws an exception that causes the client to throw as well (for example, when the client did not get the response to its request), both the service and client warning or error messages occur in the same activity for direct correlation. In the following diagram, the service throws an exception that states "The service refuses to process this request in user code." The client also throws an exception that states "The server was unable to process the request due to an internal error."  
   
- ![Using Trace Viewer to emit user&#45;code traces](../../../../../docs/framework/wcf/diagnostics/tracing/media/e2etrace2.gif "e2eTrace2")  
-Errors across endpoints for a given request appear in the same activity if the request activity id was propagated
```
Comment (mairaw) [inline docs/framework/wcf/diagnostics/tracing/emitting-user-code-traces.md]: can you move this before the image?

File: docs/framework/wcf/diagnostics/tracing/emitting-user-code-traces.md [hunk 5]
```diff
@@ -122,13 +122,11 @@ List of activities by creation time (left panel) and their nested activities (up
   
  If the service code throws an exception that causes the client to throw as well (for example, when the client did not get the response to its request), both the service and client warning or error messages occur in the same activity for direct correlation. In the following diagram, the service throws an exception that states "The service refuses to process this request in user code." The client also throws an exception that states "The server was unable to process the request due to an internal error."
```
Comment (mairaw) [inline docs/framework/wcf/diagnostics/tracing/emitting-user-code-traces.md]:
```suggestion
 If the service code throws an exception that causes the client to throw as well (for example, when the client did not get the response to its request), both the service and client warning or error messages occur in the same activity for direct correlation. In the following image, the service throws an exception that states "The service refuses to process this request in user code." The client also throws an exception that states "The server was unable to process the request due to an internal error."  
```

File: docs/framework/wcf/diagnostics/tracing/emitting-user-code-traces.md [hunk 6]
```diff
@@ -122,13 +122,11 @@ List of activities by creation time (left panel) and their nested activities (up
   
  If the service code throws an exception that causes the client to throw as well (for example, when the client did not get the response to its request), both the service and client warning or error messages occur in the same activity for direct correlation. In the following diagram, the service throws an exception that states "The service refuses to process this request in user code." The client also throws an exception that states "The server was unable to process the request due to an internal error."  
   
- ![Using Trace Viewer to emit user&#45;code traces](../../../../../docs/framework/wcf/diagnostics/tracing/media/e2etrace2.gif "e2eTrace2")  
-Errors across endpoints for a given request appear in the same activity if the request activity id was propagated  
+ ![Screenshot that shows errors across endpoints for a given request.](./media/emmiting-user-code-traces/trace-viewer-endpoint-errors.gif)
```
Comment (mairaw) [inline docs/framework/wcf/diagnostics/tracing/emitting-user-code-traces.md]: The folder where the images are has a typo as well

```suggestion
 ![Screenshot of Trace Viewer showing errors across endpoints for a given request.](./media/emitting-user-code-traces/trace-viewer-endpoint-errors.gif)  
```

File: docs/framework/wcf/diagnostics/tracing/emitting-user-code-traces.md [hunk 7]
```diff
@@ -122,13 +122,11 @@ List of activities by creation time (left panel) and their nested activities (up
...
   
- ![Using Trace Viewer to emit user&#45;code traces](../../../../../docs/framework/wcf/diagnostics/tracing/media/e2etrace2.gif "e2eTrace2")  
-Errors across endpoints for a given request appear in the same activity if the request activity id was propagated  
+ ![Screenshot that shows errors across endpoints for a given request.](./media/emmiting-user-code-traces/trace-viewer-endpoint-errors.gif)  
   
  Double-clicking the Multiply activity on the left panel shows the following graph, with the traces for the Multiply activity for each process involved. We can see a warning first occurred at the service (exception thrown), which is followed by warnings and errors on the client because the request could not be processed. Therefore, we can imply the causal error relationship between endpoints and derive the root cause of the error.  
   
- ![Using Trace Viewer to emit user&#45;code traces](../../../../../docs/framework/wcf/diagnostics/tracing/media/e2etrace3.gif "e2eTrace3")  
-Graph view of error correlation
```
Comment (mairaw) [inline docs/framework/wcf/diagnostics/tracing/emitting-user-code-traces.md]: can you move this before the image?

File: docs/framework/wcf/diagnostics/tracing/emitting-user-code-traces.md [hunk 8]
```diff
@@ -122,13 +122,11 @@ List of activities by creation time (left panel) and their nested activities (up
...
   
- ![Using Trace Viewer to emit user&#45;code traces](../../../../../docs/framework/wcf/diagnostics/tracing/media/e2etrace2.gif "e2eTrace2")  
-Errors across endpoints for a given request appear in the same activity if the request activity id was propagated  
+ ![Screenshot that shows errors across endpoints for a given request.](./media/emmiting-user-code-traces/trace-viewer-endpoint-errors.gif)  
   
  Double-clicking the Multiply activity on the left panel shows the following graph, with the traces for the Multiply activity for each process involved. We can see a warning first occurred at the service (exception thrown), which is followed by warnings and errors on the client because the request could not be processed. Therefore, we can imply the causal error relationship between endpoints and derive the root cause of the error.  
   
- ![Using Trace Viewer to emit user&#45;code traces](../../../../../docs/framework/wcf/diagnostics/tracing/media/e2etrace3.gif "e2eTrace3")  
-Graph view of error correlation  
+ ![Screenshot that shows the graph view of error correlation.](./media/emmiting-user-code-traces/trace-viewer-error-correlation.gif)
```
Comment (mairaw) [inline docs/framework/wcf/diagnostics/tracing/emitting-user-code-traces.md]:
```suggestion
 ![Screenshot of Trace Viewer showing the graph view of error correlation.](./media/emitting-user-code-traces/trace-viewer-error-correlation.gif)  
```

File: docs/standard/garbage-collection/fundamentals.md [hunk 9]
```diff
@@ -257,19 +257,17 @@ Concurrent garbage collection
   
  The following illustration shows background garbage collection performed on a separate dedicated thread on a workstation.
```
Comment (mairaw) [inline docs/standard/garbage-collection/fundamentals.md]:
```suggestion
The following illustration shows background garbage collection performed on a separate dedicated thread on a workstation:
```

File: docs/visual-basic/developing-apps/development-with-my/index.md [hunk 10]
```diff
@@ -17,7 +17,7 @@ Visual Basic provides new features for rapid application development that improv
   
  This table shows the top-level `My` objects and their relationship to each other.  
   
- ![Object Model for My](../../../visual-basic/developing-apps/development-with-my/media/myobjmodel.gif "MyObjModel")  
+ ![Diagram shows the Object Model for My.](./media/index/my-object-model-relationships.gif)
```
Comment (mairaw) [inline docs/visual-basic/developing-apps/development-with-my/index.md]:
```suggestion
 ![Diagram shows the object model for My.](./media/index/my-object-model-relationships.gif)  
```

File: docs/visual-basic/programming-guide/concepts/linq/filtering-data.md [hunk 11]
```diff
@@ -8,7 +8,7 @@ Filtering refers to the operation of restricting the result set to contain only
   
  The following illustration shows the results of filtering a sequence of characters. The predicate for the filtering operation specifies that the character must be 'A'.  
   
- ![LINQ Filtering Operation](../../../../csharp/programming-guide/concepts/linq/media/linq_filter.png "LINQ_Filter")  
+ ![Diagram that shows a LINQ Filtering Operation](./media/filtering-data/linq-filter-operation.png)
```
Comment (mairaw) [inline docs/visual-basic/programming-guide/concepts/linq/filtering-data.md]:
```suggestion
 ![Diagram that shows a LINQ filtering operation](./media/filtering-data/linq-filter-operation.png)  
```

File: docs/visual-basic/programming-guide/language-features/data-types/generic-types.md [hunk 12]
```diff
@@ -42,8 +42,7 @@ A *generic type* is a single programming element that adapts to perform the same
   
  An analogy is a screwdriver set with removable heads. You inspect the screw you need to turn and select the correct head for that screw (slotted, crossed, starred). Once you insert the correct head in the screwdriver handle, you perform the exact same function with the screwdriver, namely turning the screw.  
   
- ![Diagram of a screwdriver set as a generic tool](../../../../visual-basic/programming-guide/language-features/data-types/media/genericscrewdriver.gif "GenericScrewDriver")  
-Screwdriver set as a generic tool  
+ ![Diagram of a screwdriver set.](./media/generic-types/generic-screwdriver-set.gif)
```
Comment (mairaw) [inline docs/visual-basic/programming-guide/language-features/data-types/generic-types.md]:
```suggestion
 ![Diagram of a screwdriver set with different heads.](./media/generic-types/generic-screwdriver-set.gif)  
```

**Review by mairaw — CHANGES_REQUESTED** [review]
Thanks @damabe. A few comments before merging.

**Comment by damabe** [comment]
@mairaw I've made the changes. New CATS: https://contentqacats.azurewebsites.net/CheckResults/2019040200028/Report/

**Review by mairaw — APPROVED** [review]

**Pull request merged**